Fixes #24550 - Search repo by products

sharvit opened the pull request:
![rh repos](https://user-images.githubusercontent.com/1262502/45499234-a82eb700-b784-11e8-90d4-384ef95c9d6f.png)
![rh repos 1](https://user-images.githubusercontent.com/1262502/45499237-a95fe400-b784-11e8-9ae3-e9f06312e19a.png)

TODO:
 - [ ] Testing

sharvit commented on app/models/katello/repository.rb:
```diff
@@ -183,6 +183,8 @@ class Repository < Katello::Model
     scoped_search :on => :redhat, :complete_value => { :true => true, :false => false }, :ext_method => :search_by_redhat
     scoped_search :on => :container_repository_name, :complete_value => true
     scoped_search :on => :description, :only_explicit => true
+    scoped_search :on => :name, :relation => :product, :rename => :product_name
```
Adding `product_name` here fixes a bug that you cannot search enabled repos by `product_name`.

jlsherrill replied:
Acutally i realized that you can already search here by product name, we just rename it to 'product':

 scoped_search :rename => :product, :on => :name, :relation => :product, :complete_value => true

is that good enough, or do you actually need to search by product_name specifically?

sharvit replied:
Yes I Do do actually need to search by `product_name` specifically!

Because the same search query should work for both `available` and `enabled` repos.

sharvit commented on app/models/katello/product_content.rb:
```diff
@@ -22,6 +22,7 @@ class ProductContent < Katello::Model
     scoped_search :on => :content_type, :relation => :content
     scoped_search :on => :label, :relation => :content
     scoped_search :on => :name, :relation => :product, :rename => :product_name
+    scoped_search :on => :id, :relation => :product, :rename => :product_id
```
Added scope search by `product_id` to optimize the search.

jlsherrill commented on app/models/katello/repository.rb:
```diff
@@ -183,6 +183,8 @@ class Repository < Katello::Model
     scoped_search :on => :container_repository_name, :complete_value => true
     scoped_search :on => :description, :only_explicit => true
+    scoped_search :on => :name, :relation => :product, :rename => :product_name
+    scoped_search :on => :id, :relation => :product, :rename => :product_id
```
these look fine, but you do you mind adding a test for them?  It should be fairly simple, we generally test all of our searches.

Rohoover commented:
@sharvit 
Looks good so far, how big is that dropdown (how long is the list)?

waldenraines commented:
> Looks good so far, how big is that dropdown (how long is the list)?

@Rohoover it's going to be a list of all products.  We can iterate on this and add dropdown with search capabilities afterwords.

Rohoover commented:
> > Looks good so far, how big is that dropdown (how long is the list)?
> 
> @Rohoover it's going to be a list of all products. We can iterate on this and add dropdown with search capabilities afterwords.

That would be perfect. Guarantee 100% customers will ask about that.

waldenraines commented:
Also, I wonder if we should update the multi-select to allow passing a prop to display the "4 items selected" text when more than two are selected since the product names are really long and make the view look weird when multiple products are selected:

![gscreenshot_2018-09-14-095733](https://user-images.githubusercontent.com/4116405/45554721-056a4d00-b805-11e8-9a20-392b77d5fcf0.png)

Rohoover commented:
@waldenraines 
Agreed. Also moving the CSV button will stop that huge unsightly gap there.

sharvit commented:
Thanks for the reviews :+1: 
I updated everything you mention in your comments.

@jlsherrill - I added those tests but I couldn't ran them locally.
Somehow I am getting this error when running tests:
https://gist.github.com/sharvit/93f4af13581fb578336716046dce9393

Still need to finish the js-tests

sharvit commented:
@jlsherrill - Seems like the new rb tests works in travis :)

@waldenraines - JS tests added.

waldenraines commented:
[test katello]

waldenraines commented:
[test katello]

waldenraines commented:
[test katello]

waldenraines commented:
[test katello]

waldenraines commented:
@sharvit I think @Rohoover wanted the `Export as CSV` button inside the grey area of the enabled repos section with a similar position to the recommended repositories selector on the available repos section.

jlsherrill commented:
@sharvit ahh okay, not sure why the tests through that error for you, but they look fine and pass here.  ACK from a ruby perspective.

Rohoover commented:
Including the screenshot for reference.

<img width="838" alt="screen shot 2018-09-20 at 3 50 08 pm" src="https://user-images.githubusercontent.com/18266453/45843380-702bf480-bced-11e8-8458-e98bb9007a33.png">

waldenraines commented:
[test katello]

waldenraines commented:
Closing in favor of https://github.com/Katello/katello/pull/7717.
